sdk/src/server/mod.rs

```diff
@@ -18,10 +23,38 @@ use self::types::{
     AggregationCircuitCtx, AxiomComputeCircuitCtx, AxiomComputeCtx, AxiomComputeJobStatus,
     AxiomComputeManager, AxiomComputeServerCmd,
 };
-use crate::utils::io::{read_agg_pk_and_pinning, read_metadata, read_pinning, read_pk};
+use crate::{
+    compute::{AxiomCompute, AxiomComputeFn},
+    utils::io::{read_agg_pk_and_pinning, read_metadata, read_pinning, read_pk},
+};
 
 pub mod types;
 
+pub trait AxiomClientProvingServer: AxiomCircuitScaffold<Http, Fr> {
+    type Context = ();
+    type RequestInput: DeserializeOwned;
+
+    fn construct_context(params: Self::CoreParams) -> Self::Context;
+    fn process_input(ctx: Self::Context, input: Self::RequestInput) -> Self::InputValue;
+    #[allow(unused_variables)]
+    fn after_prove(ctx: Self::Context, output: AxiomV2CircuitOutput) {}
+}
+
+impl<A: AxiomComputeFn> AxiomClientProvingServer for AxiomCompute<A>
```

Comment:
This means you can't override it for `AxiomCompute`? Can we just leave it unimplemented? or else do `A: AxiomComputeFn + AxiomCompute`, so it can be implemented there

Reply:
We can address this later